tests/main/mount-ns/google.ubuntu-16.04-64/HOST.expected.txt

```diff
@@ -19,7 +19,7 @@
 2:3 / /snap/test-snapd-mountinfo-core16/1 ro,nodev,relatime shared:19 - squashfs /dev/loop3 ro
 2:4 / /snap/test-snapd-mountinfo-core18/1 ro,nodev,relatime shared:20 - squashfs /dev/loop4 ro
 1:14 / /sys rw,nosuid,nodev,noexec,relatime shared:21 - sysfs sysfs rw
-1:15 / /sys/fs/cgroup rw shared:22 - tmpfs tmpfs rw,mode=755
+1:15 / /sys/fs/cgroup ro,nosuid,nodev,noexec shared:22 - tmpfs tmpfs ro,mode=755
```

Comment:
I think we're missing something here. The inherit variant of the test fails, but the reboot one is successful, so some earlier test must have caused the changes to the mount namespace. I will try to investigate which test might have caused it.

Reply:
After more digging I've determined the following:
- clean 16.04 host comes up with `/sys/fs/cgroup` mounted in `rw` mode
- running `sysctemctl daemon-reexec` switches the mode to `ro,nosuid..`
- the tests upgrade systemd as it's a dependency for snapd
- systemd.postinst runs `_systemctl daemon-reexec || true` which triggers the change

So far it is unclear why the clean system has `rw` mode, AFAIU it should already be `ro` after reboot.

Reply:
Another data point:
- cloud images from cloud-images.u.com have `ro,nosuid..` mount options
- after installing `linux-image-gcp` on xenial cloud image, the mounts are still correct

Reply:
And cannot reproduce this outside of GCP. Attempted installing gcp services in a VM, nothing.

Booting GCP image with systemd debugs enabled, does not yield anything useful either. The place in the code where the mounts happen is in early daemon init and there's no specific logging there. Also, there were no other logs suggesting that the /sys/fs/cgroup mountpoint got changed.

Looking at systemd code, /sys/fs/cgroup is remounted always when systemd starts up, so it's not something that was left behind in this weird state by initramfs.